controllers/gce/controller/cluster_manager.go

```diff
@@ -212,17 +212,28 @@ func (c *ClusterManager) GC(lbNames []string, nodePorts []backends.ServicePort)
 }
 
 func getGCEClient(config io.Reader) *gce.GCECloud {
-	allConfig, err := ioutil.ReadAll(config)
-	if err != nil {
-		glog.Fatalf("Error while reading entire config: %v", err)
+	getConfigReader := func() io.Reader { return nil }
+
+	if config != nil {
+		allConfig, err := ioutil.ReadAll(config)
+		if err != nil {
+			glog.Fatalf("Error while reading entire config: %v", err)
+		}
+		glog.V(2).Infof("Using cloudprovider config file:\n%v ", string(allConfig))
+
+		getConfigReader = func() io.Reader {
```

Comment:
Not quite that simple. If we went this route, `configReader` would need to be reset for each loop iteration.